### PR TITLE
Next (v0.25.0)

### PR DESCRIPTION
**Pull request by nollety**
# Description

This is the follow-up of https://github.com/eradiate/eradiate/pull/344.

Additional changes are:

* Fix: compatible geometry must be set
* Bump `joseki` to `joseki>=2.5.0`
* Add `eval_transmittance_*` methods
* Add unit test for experiment with CAMS atmosphere
* `QuadratureSpecifications`: fixes and improvements
* RTLS BSDF
* various other fixes, refactorings and improvements related to circular imports, the command-line-interface, data download lists, pipelines, documentation

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is appropriately documented
- [x] The code is tested to prove its function
- [x] The feature branch is rebased on the current state of the `main` branch
- [x] I updated the change log if relevant
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

### CODE REVIEW

**Comment by nollety** [comment]
@leroyvn, just to let you know that the scope of this PR is going beyond that of the custom atmospheric profiles ; the RTLS BSDF is part of it too.

**Comment by nollety** [comment]
All tests pass except `tests/01_eradiate/02_system/test_atmosphere_rpv.py::test_rpv_vs_lambertian[0.5-homogeneous]` : Mismatched elements: 1 / 14 (7.14%).

File: docs/rst/maintainer_guide.rst
Comment (leroyvn) [reply docs/rst/maintainer_guide.rst:289]: I do, unless changes @wint3ria made would have made this irrelevant.

**Comment by wint3ria** [comment]
I upgraded the kernel to include the latest RTLS patch. I'd like to make a RC on PyPI and test it internally before we merge. I am in the process of testing my few commits.

File: docs/rst/maintainer_guide.rst
Comment (wint3ria) [reply docs/rst/maintainer_guide.rst:289]: yes it is relevant in the sense that the lock files must be updated, but dependencies are listed in the layered.yml file now... I'll update the docs. My latest commits focus on updating the lock files anyway :)

Comment (wint3ria) [reply docs/rst/maintainer_guide.rst:289]: Ah you did update the docs. Sorry did not see it at first. Great :)

**Comment by wint3ria** [comment]
I can reproduce the failing test on RPV. I sometimes have a failure on the reflectance value of 0.5, 0, or no failures at all. This is a known issue in our test suite, but I'll make an effort to alleviate the problem for now

**Comment by wint3ria** [comment]
The RC in compiling

**Comment by wint3ria** [comment]
RC available [here](https://pypi.org/project/eradiate/0.25.0rc8/)

**Comment by nollety** [comment]
I managed to refactor the code so that the `portion` library is not used/required anymore. Some subtle differences exist between the way SRFs (`InterpolatedSpectrum` or `MultiDeltaSpectrum`) selects wavelengths and wavelength bins now compared to before (#344). All tests have passed on this last commit (which is amazing).

**Comment by nollety** [comment]
> I upgraded the kernel to include the latest RTLS patch. I'd like to make a RC on PyPI and test it internally before we merge. I am in the process of testing my few commits.

@wint3ria what's the status on the RTLS testing, would you like to test it further or is it good to go?

File: src/eradiate/radprops/absorption.py
Comment (leroyvn) [inline src/eradiate/radprops/absorption.py:683]: Maybe expand `kw` to a more explicit name?

**Review by leroyvn — CHANGES_REQUESTED** [review]
Hi @nollety, here are a few specific comments (some to be addressed by @wint3ria). I didn't comment too much on the changes you made to the post-processing pipelines, these are anyway being completely rewritten.

File: tests/01_eradiate/03_regression/atmospheres/test_rpv_afgl1986_continental.py
Comment (nollety) [reply tests/01_eradiate/03_regression/atmospheres/test_rpv_afgl1986_continental.py:82]: Because this is a regression test and I thought setting explicitly this config would be safer than using some default defined elsewhere, since in regression test we are concerned with the problem that the same input should give the same output.

File: tests/01_eradiate/03_regression/atmospheres/test_rpv_afgl1986.py
Comment (nollety) [reply tests/01_eradiate/03_regression/atmospheres/test_rpv_afgl1986.py:78]: Ditto

File: src/eradiate/data/_store.py
Comment (nollety) [reply src/eradiate/data/_store.py:68]: Code readability

File: tests/01_eradiate/01_unit/scenes/atmosphere/test_heterogeneous.py
Comment (nollety) [reply tests/01_eradiate/01_unit/scenes/atmosphere/test_heterogeneous.py:90]: I think so yes

File: tests/01_eradiate/01_unit/radprops/test_rayleigh_scattering.py
Comment (nollety) [reply tests/01_eradiate/01_unit/radprops/test_rayleigh_scattering.py:67]: What do you mean? Are you referring to the commented line?

File: src/eradiate/spectral/ckd.py
Comment (nollety) [reply src/eradiate/spectral/ckd.py:174]: yes

File: src/eradiate/scenes/atmosphere/_molecular.py
Comment (nollety) [reply src/eradiate/scenes/atmosphere/_molecular.py:67]: This could be done in a later update

File: src/eradiate/scenes/atmosphere/_molecular.py
Comment (nollety) [reply src/eradiate/scenes/atmosphere/_molecular.py:147]: Fixed

File: CHANGELOG.md
Comment (nollety) [reply CHANGELOG.md:109]: Completely agree

File: src/eradiate/radprops/absorption.py
Comment (nollety) [reply src/eradiate/radprops/absorption.py:683]: I find it explicit enough, but feel free to rename it to your liking :)

File: src/eradiate/kernel/__init__.py
Comment (nollety) [reply src/eradiate/kernel/__init__.py:5]: Yep

**Comment by nollety** [comment]
All tests pass on my end.

**Comment by nollety** [comment]
I rebased on the latest main branch

**Comment by nollety** [comment]
`eradiate show` produces these warnings:

```shell
• Using an incompatible version of Mitsuba. Eradiate requires Mitsuba 3.2.1. Found Mitsuba 3.4.0.
• Using an incompatible patch version of Mitsuba. Eradiate requires a Mitsuba kernel version 3.2.1, with a specific patch version 0.1.1rc1. Found Mitsuba 3.4.0 with the patch version 0.1.1.
```
I am not sure how to deal with them

**Comment by nollety** [comment]
@leroyvn this is ready for another review if you wish. Also, shall we wait for @wint3ria's review? Let me know when I can squash the commits.

**Comment by leroyvn** [comment]
> `eradiate show` produces these warnings:
> 
> ```shell
> • Using an incompatible version of Mitsuba. Eradiate requires Mitsuba 3.2.1. Found Mitsuba 3.4.0.
> • Using an incompatible patch version of Mitsuba. Eradiate requires a Mitsuba kernel version 3.2.1, with a specific patch version 0.1.1rc1. Found Mitsuba 3.4.0 with the patch version 0.1.1.
> ```
> 
> I am not sure how to deal with them

My guess is: you need to recompile the kernel.

**Comment by wint3ria** [comment]
I did not get involved in the atmospheric part of this PR, I only use the API. I can tell that for the number of RAMI4ATM cases using RTLS that I could afford to check manually against the references I had, the results are not "too crazy". Without a more trust worthy testing facility, I don't think we can do much more. I can rerun a batch of non-RTLS comparison cases if you'd like, but I won't do it before the end of the week.

**Comment by wint3ria** [comment]
I've proposed a solution for the path expansion issue, let me know what you think

File: docs/src/release_notes/v0.25.x.md
Comment (wint3ria) [inline docs/src/release_notes/v0.25.x.md:6]: *a better flexibility

File: docs/src/release_notes/v0.25.x.md
Comment (wint3ria) [inline docs/src/release_notes/v0.25.x.md:8]: Let's move this to the "added" section

File: docs/src/release_notes/v0.25.x.md
Comment (wint3ria) [inline docs/src/release_notes/v0.25.x.md:77]: To add : fix the data path in the store

File: ext/mitsuba
Comment (wint3ria) [inline ext/mitsuba:1]: We need to update the submodule again

File: requirements/pip/dependencies.in
Comment (wint3ria) [inline requirements/pip/dependencies.in:1]: We need to use eradiate-mitsuba 0.1.1, when it will be delivered

File: requirements/pip/optional.in
Comment (wint3ria) [inline requirements/pip/optional.in:1]: same comment

**Review by wint3ria — COMMENTED** [review]
We're not far from merging. I will perform some further testing and update the necessary versions

File: ext/mitsuba
Comment (wint3ria) [reply ext/mitsuba:1]: This is not necessary, since we droped the support for Windows for the version 0.25.0

**Comment by wint3ria** [comment]
#373 We delay the Windows support for now. We'll have it in an upcoming patch

**Comment by wint3ria** [comment]
run Eradiate CI

**Pull request merged**
